### PR TITLE
Configured CI to build all packages

### PR DESCRIPTION
**Pull request by palczyn**
Issues: KNTK-368
Based on PR #74

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -65,3 +65,46 @@ jobs:
         uses: actions/checkout@v2
       - name: Check Go module consistency
         run: ./tools/check_go_mod.sh
+
+  build:
```
Comment (danielfurman) [inline .github/workflows/test.yml]: Please move it above _test_ job and run tests only when build succeeds.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+    name: Build binary
```
Comment (danielfurman) [inline .github/workflows/test.yml]:
```suggestion
    name: Build code
```
We're building multible non-executable binary object files here so name _binary_ doesn't sound right for me.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+    runs-on: 'ubuntu-latest'
```
Comment (danielfurman) [inline .github/workflows/test.yml]:
```suggestion
    runs-on: ubuntu-latest
```
FYI there is no need for `'` nor `"` for YAML strings

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+    needs: test
+    strategy:
+      matrix:
+        goosarch:
+          - 'linux/386'
```
Comment (danielfurman) [inline .github/workflows/test.yml]: As discussed, please use linux/amd64, windows/amd64, darwin/amd64, darwin/arm

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+          - 'linux/ppc64'
+          - 'linux/ppc64le'
+          - 'linux/riscv64'
+          - 'linux/s390x'
+          - 'windows/386'
+          - 'windows/amd64'
+          - 'windows/arm'
+        go-version:
+          - 1.15.x
+          - 1.16.x
```
Comment (danielfurman) [inline .github/workflows/test.yml]: As discussed, please bump go versions up to 1.17

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2.4.0
+      - uses: actions/setup-go@v2
```
Comment (danielfurman) [inline .github/workflows/test.yml]: Nit: please name it `- name: Install Go`

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -65,3 +65,46 @@ jobs:
...
+        run: |
+          GOOSARCH=${{matrix.goosarch}}
+          GOOS=${GOOSARCH%/*}
+          GOARCH=${GOOSARCH#*/}
+          echo "GOOS=$GOOS" >> $GITHUB_ENV
+          echo "GOARCH=$GOARCH" >> $GITHUB_ENV
+      - name: Build
+        run: |
+          go build -v ./...
+          go build -tags examples ./...
```
Comment (danielfurman) [inline .github/workflows/test.yml]: As discussed, this command compiler all code and is sufficent by itself

**Review by danielfurman — CHANGES_REQUESTED** [review]

**Review by mateuszmidor — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 2 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 3 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 4 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 5 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 6 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

File: .github/workflows/test.yml (hunk 7 above)
Comment (danielfurman) [reply .github/workflows/test.yml]: Done

**Review by danielfurman — APPROVED** [review]

**Review by mmac-m3a — COMMENTED** [review]
I'll hold on reviewing this until previous PRs are merged and this change rebased, so I don't have to go over the same files again.

**Review by mmac-m3a — APPROVED** [review]

**Pull request merged**